icu4j/main/classes/core/src/com/ibm/icu/message2/functions/DateTimeFormatterFactory.java

```diff
@@ -0,0 +1,120 @@
+// © 2022 and later: Unicode, Inc. and others.
+// License & terms of use: http://www.unicode.org/copyright.html
+
+package com.ibm.icu.message2.functions;
+
+import java.util.Locale;
+import java.util.Map;
+import java.util.Objects;
+
+import com.ibm.icu.message2.FormattedPlaceholder;
+import com.ibm.icu.message2.Formatter;
+import com.ibm.icu.message2.FormatterFactory;
+import com.ibm.icu.message2.PlainStringFormattedValue;
+import com.ibm.icu.text.DateFormat;
+import com.ibm.icu.text.SimpleDateFormat;
+
+/**
+ * Creates a {@link Formatter} doing formatting of date / time, similar to
+ * <code>{exp, date}</code> and <code>{exp, time}</code> in {@link com.ibm.icu.text.MessageFormat}.
+ *
+ * @internal ICU 72 technology preview
+ * @deprecated This API is for ICU internal use only.
+ */
+@Deprecated
+public class DateTimeFormatterFactory implements FormatterFactory {
+
+    private static int stringToStyle(String option) {
+        switch (option.toUpperCase(Locale.US)) {
+            case "FULL": return DateFormat.FULL;
+            case "LONG": return DateFormat.LONG;
+            case "MEDIUM": return DateFormat.MEDIUM;
+            case "SHORT": return DateFormat.SHORT;
+            default: return DateFormat.DEFAULT;
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * @internal ICU 72 technology preview
+     * @deprecated This API is for ICU internal use only.
+     */
+    @Deprecated
+    @Override
+    public Formatter createFormatter(Locale locale, Map<String, Object> fixedOptions) {
+        DateFormat df;
+
+        Object opt = fixedOptions.get("skeleton");
+        if (opt != null) {
+            String skeleton = Objects.toString(opt);
+            df = DateFormat.getInstanceForSkeleton(skeleton, locale);
+            return new DateTimeFormatter(df);
```

Comment:
It is function dependent.
I would expect that part of the function "registration" in the standard to describe the options, priorities, etc.
Since the functions are not standard, I made up my own rules (skeleton takes priority).

Reply:
TODO: how to handle conflicts

Reply:
Not a blocker

Reply:
Done
Added TODO in `DateTimeFormatterFactory`.
Although the decision is not date-time specific, it is general, for all formatters.